framework/src/modules/token/method.ts

```diff
@@ -706,8 +746,49 @@ export class TokenMethod extends BaseMethod {
 		this.events.get(TokenIDSupportedEvent).log(methodContext, tokenID);
 	}
 
-	public async removeSupportTokenID(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
-		await this.stores.get(SupportedTokensStore).removeSupportForToken(methodContext, tokenID);
+	public async removeSupport(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
+		const [chainID] = splitTokenID(tokenID);
+
+		const allTokensSupported = await this.stores
+			.get(SupportedTokensStore)
+			.has(methodContext, ALL_SUPPORTED_TOKENS_KEY);
+
+		if (allTokensSupported) {
+			throw new Error('All tokens are supported.');
+		}
+
+		if (tokenID.equals(this.getMainchainTokenID()) || chainID.equals(this._config.ownChainID)) {
+			throw new Error('Cannot remove support for the specified token.');
+		}
+
+		try {
```

Comment:
```
const chainSupported = await this.stores.get(SupportedTokensStore).has(methodContext, chainID);
if (!chainSupported) {
	this.events.get(TokenIDSupportRemovedEvent).log(methodContext, tokenID);
	return;
}
```
it's better to use `has` function than handling error here as well. Also, from the LIP i think it suppose to log the event in case chain is not supported as well

Reply:
@shuse2, according to the LIP, an event is `only` emitted on successful removal. An error is thrown instead `if (!chainSupported) `

Reply:
Looking at https://github.com/LiskHQ/lips/blob/main/proposals/lip-0051.md#removesupport
```
if supportedTokens(chainID) exists:
```
and 
```
emitEvent(
        module=MODULE_NAME_TOKEN,
        name=EVENT_NAME_TOKEN_ID_SUPPORT_REMOVED,
        data={"tokenID": tokenID},
        topics=[tokenID]
    )
```
are in the same indentation. so it looks like if chain is not supported, it will still log the event